lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/ApiSpec.hs

```diff
@@ -11,22 +12,107 @@ module Cardano.Wallet.Jormungandr.ApiSpec
+
+            it "With 0 stake pools" $ do
+
+                let testAccountState = Aeson.encode [aesonQQ|
+                        { "value": #{testBalance}
+                        , "counter": #{testTransactionCount}
+                        , "delegation": {"pools": []}
+                        }|]
+                eitherDecode testAccountState `shouldBe` Right ApiAccountState
+                    { currentBalance =
+                        ApiT (Quantity testBalance)
+                    , totalTransactionCount =
+                        ApiT (Quantity testTransactionCount)
+                    , delegationInfo = ApiAccountDelegationInfo
+                        { stakePools = [] }
+                    }
+
+            it "With 1 stake pool" $ do
+
+                let testAccountState = Aeson.encode [aesonQQ|
+                        { "value": #{testBalance}
+                        , "counter": #{testTransactionCount}
+                        , "delegation":
+                            { "pools":
+                                [ [#{toText testPoolId1}, #{testPoolRatio1}] ]
+                            }
+                        }|]
+                eitherDecode testAccountState `shouldBe` Right ApiAccountState
+                    { currentBalance =
+                        ApiT (Quantity testBalance)
+                    , totalTransactionCount =
+                        ApiT (Quantity testTransactionCount)
+                    , delegationInfo = ApiAccountDelegationInfo
+                        { stakePools =
+                            [ ( ApiT testPoolId1
+                              , ApiT (Quantity testPoolRatio1))
+                            ]
+                        }
+                    }
+
+            it "With n stake pools" $ do
+
+                let testAccountState = Aeson.encode [aesonQQ|
+                        { "value": #{testBalance}
+                        , "counter": #{testTransactionCount}
+                        , "delegation":
+                            { "pools":
+                                [ [#{toText testPoolId1}, #{testPoolRatio1}]
+                                , [#{toText testPoolId2}, #{testPoolRatio2}]
+                                ]
+                            }
+                        }|]
+                eitherDecode testAccountState `shouldBe` Right ApiAccountState
+                    { currentBalance =
+                        ApiT (Quantity testBalance)
+                    , totalTransactionCount =
+                        ApiT (Quantity testTransactionCount)
+                    , delegationInfo = ApiAccountDelegationInfo
+                        { stakePools =
+                            [ ( ApiT testPoolId1
+                              , ApiT (Quantity testPoolRatio1))
+                            , ( ApiT testPoolId2
+                              , ApiT (Quantity testPoolRatio2))
+                            ]
+                        }
+                    }
```

Comment:
What about using `QuickCheck` + `Arbitrary` here to perform those decoding? This is clearly a case where property based testing shines -> generating random test cases trying to find edge-cases, rather than having a possibly non-exhaustive list of unit test. 

Usually, my own preferences is to have a mix of both QC and "standard unit tests". Having simply a one or a couple of hand-written test case like the above to really convince myself that a particular things follow my intuition, and then, hammer it down with QC.

Reply:
@KtorZ wrote:
> What about using `QuickCheck` + `Arbitrary` here to perform those decoding? This is clearly a case where property based testing shines -> generating random test cases trying to find edge-cases, rather than having a possibly non-exhaustive list of unit test.

Fair enough! I've added roundtrip JSON encoding/decoding tests for the `AccountState` type.

I think it's still worth keeping the handwritten positive unit tests, as these are directly copied from example JSON output provided by JCLI.

Reply:
> I think it's still worth keeping the handwritten positive unit tests, as these are based on actual examples of JSON output from the jcli tool.

Yes, hence my second point in my original comment. Having a couple of hand-written scenario is useful to build a correct intuition of a particular behavior. But the actual testing is done by QC ^.^